src/lang/base/BuiltIns.ml

```diff
@@ -1152,12 +1151,12 @@ module ScillaBuiltIns
             pure (type_elab, res_type, exec)
           else fail0 @@ "Name or arity don't match") in
       let open Caml in
-      let dict = Option.value ~default:[] @@ Hashtbl.find_opt built_in_hashtbl opname in
+      let dict = Option.value ~default:[] @@ Hashtbl.find_opt built_in_hashtbl op in
       let%bind (_, (type_elab, res_type, exec)) = tryM dict ~f:finder
           ~msg:(fun () ->
               mk_error1
-                (sprintf "Cannot find built-in with name \"%s\" and argument types %s." opname (pp_typ_list argtypes))
-                (ER.get_loc (get_rep op)))
+                (sprintf "Cannot find built-in with name \"%s\" and argument types %s." (pp_builtin op) (pp_typ_list argtypes))
```

Comment:
Since we rule out incorrect builtin names on the parsing stage, it'd clearer if the error message said something along the following lines:
```
Type error: cannot apply \"%s\" built-in to argument(s) of type(s) %s.
```